src/ScalarPotentials/ScalarPotential.jl

```diff
@@ -48,14 +48,15 @@ function get_2π_potential(sp::ScalarPotential{T, 3, Cylindrical}, axφ::Discret
     for idx_n in 1:n
         for il in 1:l
             idx::Int = il + (idx_n - 1) * l
-            new_ticks[idx] = (idx_n - 1) * Δφ + axφ.ticks[il]
+            new_ticks[idx] = mod((idx_n - 1) * Δφ + axφ.ticks[il], T(2π))
             new_pot[:, idx, :] = sp[:, il, :]
         end
     end
-    new_axφ = DiscreteAxis{AT, :periodic, :periodic}( new_int, new_ticks )
+    P = sortperm(new_ticks)
+    new_axφ = DiscreteAxis{AT, :periodic, :periodic}( new_int, new_ticks[P] )
     new_axes = (sp.grid[1], new_axφ, sp.grid[3])
     new_grid = Grid{AT, 3, Cylindrical, typeof(new_axes)}( new_axes )
-    ScalarPotential(sp, new_pot, new_grid)
+    ScalarPotential(sp, new_pot[:,P,:], new_grid)
```

Comment:
Also here, `g.φ.ticks[1]` is not necessarily equal to `g.φ.interval.left`.
Please compose an example where `g.φ.ticks` does not include `0` or multiples of `2pi` (probably a 2D symmetric detector from `phi: [17°,17°]` or something like this and check whether the expansion of the potential to full `2pi` affects the calculation, plotting, etc.
It would also be good to add such a case with a non-zero left `phi` entry for the `grid` in the config file to the tests.

Reply:
I already added such a geometry and will write tests. Though when using a 2D geometry we only test `extend_2D_to_3D_by_n_points`. We are not testing `get_2π_potential` for periodic boundary conditions. What is not yet implemented is `get_2π_potential` for reflecting boundaries, which should also be tested.